gnomad_constraint/plots/loading_functions.R

```diff
@@ -0,0 +1,285 @@
+library(conflicted)
+library(dplyr)
+library(gargle)
+library(googleAuthR)
+library(googleCloudStorageR)
+library(forcats, include.only = c("fct_recode"))
+library(purrr, include.only = c("map_df"))
+library(readr, include.only = c("read_tsv", "cols"))
+library(stringr, include.only = c("str_sub"))
+
+conflict_prefer("filter", "dplyr")
+conflict_prefer("select", "dplyr")
+conflict_prefer("mutate", "dplyr")
+conflict_prefer("%>%", "dplyr")
+
+####################################################################
+# Constants and functions related to data loading
+####################################################################
+data_versions <- c("v2", "v4")
+map_data_version <- list(
+  v2 = "2.1.1",
+  v4 = "4.1",
+  v4_tmp = "4.1"
+)
+
+default_output_path <- "."
+default_data_dir <- "data"
+default_plot_dir <- "plots"
+default_gene_lists <- "data/gene_lists"
+bucket <- "gs://gnomad"
+
+setup_directories <- function(
+    output_path = default_output_path,
+    data_dir = default_data_dir,
+    plot_dir = default_plot_dir,
+    gene_lists = default_gene_lists) {
+  # Create the output directory and subdirectories if they don't exist
+  # output_path: The base directory for the output
+  # data_dir: The subdirectory for data files
+  # plot_dir: The subdirectory for plot files
+  # gene_lists: The subdirectory for gene list files
+  # Returns: None
+  suppressWarnings(dir.create(output_path))
+  suppressWarnings(dir.create(sprintf("%s/%s/", output_path, data_dir)))
+  suppressWarnings(dir.create(sprintf("%s/%s/", output_path, plot_dir)))
+  suppressWarnings(dir.create(sprintf("%s/%s/", output_path, gene_lists)))
+}
+
+get_data_url <- function(version = "v4", release = TRUE, public = TRUE) {
+  # Get the URL for the data files
+  # version: The version of the data
+  # release: Whether to use the release or development version
+  # public: Whether to use the public or non-public version
+  # Returns: The URL for the data files
+
+  # TODO: Modify if there are changes for v4.1
+  full_version <- map_data_version[[version]]
+  if (release && public) {
+    data_path <- sprintf(
+      "gs://gcp-public-data--gnomad/release/%s/constraint/",
+      full_version
+    )
+  } else if (version == "v4_tmp") {
+    data_path <- "gs://gnomad-kristen/constraint/"
+  } else if (release && !public) {
+    data_path <- sprintf("gs://gnomad/release/%s/constraint/", full_version)
+  } else if (!release && !public) {
```

Comment:
i use the old v2 version for plotting rather than the reformatted/productionized, so not all these paths will pull the old version?

Reply:
Yeah, when it's called in the main script it is 
```v2_constraint_data <- load_constraint_metrics(
  version = "v2",
  output_path = output_path
)
```

which uses public = TRUE and release = TRUE as default